Tests/Unit/FrameworkTest.php

```diff
@@ -5346,4 +5347,34 @@ public function createBackEndUserGroupForTitleGivenStoresTitleInGroupRecord()
             )
         );
     }
+
+    /**
+     * Populate $GLOBALS['TYPO3_DB'] and create test database
+     *
+     * @see https://github.com/Nimut/testing-framework/blob/master/src/TestingFramework/TestSystem/TestSystem.php
+     *
+     * @throws Exception
+     * @throws \ReflectionException
+     * @return void
+     */
+    protected function setUpTestDatabase()
+    {
+        if (VersionNumberUtility::convertVersionNumberToInteger(TYPO3_version) >= 8001000) {
+            // The TYPO3 core misses to reset its internal connection state
+            // This means we need to reset all connections to ensure database connection can be initialized
+            $closure = \Closure::bind(function () {
+                foreach (ConnectionPool::$connections as $connection) {
+                    $connection->close();
+                }
+                ConnectionPool::$connections = [];
+            }, null, ConnectionPool::class);
+            $closure();
+
+            // get database name from $GLOBALS['TYPO3_DB']
+            $dbName = $this->getProtectedProperty($GLOBALS['TYPO3_DB'], 'databaseName');
+
+            // set correct database name for doctrine
+            $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections']['Default']['dbname'] = $dbName;
```

Comment:
We need to think about this, the `Connections` doesn't need to be named `Default` is does in our case, but could be called what ever we like..

Reply:
Yes correct. But this extensions setups the database connection so it is in "our hand" to decide to always use "Default". I can use the class constant in ConnectionPool to make it more nicer ;-)